File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        # prev,curr,next
        prev,curr = None, head
        result = prev

        while head is not None:
            curr = head.next
            head.next = prev
            prev = head
            head = curr
        
        print(prev)
        return prev

File: Data Structures & Algorithms/linked-list-cycle-detection/submission-0.kt
/**
 * Definition for singly-linked list.
 * class ListNode(var `val`: Int) {
 *     var next: ListNode? = null
 * }
 */

class Solution {
    fun hasCycle(head: ListNode?): Boolean {
        var fastPointer = head?.next?.next
        var currentPointer = head

        while(currentPointer?.next != null) {
            if(fastPointer == currentPointer) return true
            fastPointer = fastPointer?.next?.next
            currentPointer = currentPointer?.next
        }
        return false
    }
}
